src/server/frontend_xwayland/xwayland_client_manager.cpp

```diff
@@ -80,6 +84,11 @@ auto mf::XWaylandClientManager::session_for_client(pid_t client_pid) -> std::sha
     }
     else
     {
+        if (!session_authorizer->connection_is_allowed({client_pid, getuid(), getgid()}))
```

Comment:
Assuming `client_pid` is sensible, there ought to be a way to get gid/uid from that. This seems clunky but...

```c++
struct stat proc_stat;
stat(std::format("/proc/{}", client_pid).c_str(), &proc_stat);
auto const uid = proc_stat.st_uid;
auto const gid = proc_stat.st_gid;
```

Reply:
I was considering the exact same thing, so i'll try to add this.

Reply:
Implemented this, but did not use std::format as that require gcc 13++